src/main/javassist/util/proxy/ProxyFactory.java

```diff
@@ -1027,18 +1027,24 @@ private int overrideMethods(ClassFile cf, ConstPool cp, String className, ArrayL
             Map.Entry e = (Map.Entry)it.next();
             String key = (String)e.getKey();
             Method meth = (Method)e.getValue();
-            if (ClassFile.MAJOR_VERSION < ClassFile.JAVA_5 || !isBridge(meth))
```

Comment:
if `isBridge(method)` returns false then we have also fixed the problem.

Is this truly a scenario where method is really bridged but public and overloaded has per this solution.

Reply:
I did not want to touch isBridge(method) due to this method is simple and it reflects if a method is actual Bridge or not. And if a method is Bridge or not is decided by Java compiler. If we change it, then confusion will be introduced in the future.